cypress/e2e/API/EngageSphere.cy.js

```diff
@@ -0,0 +1,151 @@
+/// <reference types="Cypress" />
+describe('Validações de API', () => {
+    context('Solicitações inválidas de forma elegante', () => {
+      it('Então valido page="invalid" como string', () => {
+        cy.request({
+          method: 'GET',
+          url: 'http://localhost:3001/customers?page=invalid&limit=10&size=All',
+          failOnStatusCode: false 
+        }).then(response => {
+          expect(response.status).to.eq(400)
+        });
+      });
+      it('Então valido a request definindo como extra_large', () => {
+        cy.request({
+          method: 'GET',
+          url: 'http://localhost:3001/customers?size=extra_large',
+          failOnStatusCode: false
+        }).then((response) => {
+          expect(response.status).to.eq(400);
+          expect(response.body.error).to.contain('Unsupported size value');
+          expect(response.body.error).to.contain('Supported values are All, Small, Medium, Enterprise, Large Enterprise, and Very Large Enterprise.');
+        });
+      });
+      it('Então valido a request definindo como invalid limit', () => {
+        cy.request({
+          method: 'GET',
+          url: 'http://localhost:3001/customers?page=invalid&limit=10&size=All',
+          failOnStatusCode: false 
+        }).then(response => {
+          expect(response.status).to.eq(400);
+        });
+      });
+      it('Então valido a request com tamanho invalido', () => {
+        cy.request({
+          method: 'GET',
+          url: 'http://localhost:3001/customers?page=1&limit=10&size=Invalid',
+          failOnStatusCode: false
+        }).then(response => {
+          expect(response.status).to.eq(400);
+        });
+      })
+      it('Então valido limit como boleano', () => {
+        cy.request({
+          method: 'GET',
+          url: 'http://localhost:3001/customers?page=1&limit=true&size=Large%20Enterprise',
+          failOnStatusCode: false
+        }).then(response => {
+          expect(response.status).to.eq(400);
+          //Obs -> não retorna mensagem
+        });
+      });
+    });
+    context('Validação de paginação', () => {
+        it('Validação de paginação da primeira página', () => {
+          cy.request({
+            method: 'GET',
+            url: 'http://localhost:3001/customers?page=1&limit=10&size=All'
+          }).then((response) => {
+            console.log('Clientes da primeira página:', response.body);
+            expect(response.status).to.eq(200);
+            expect(response.body.customers.length).to.eq(10);
+            expect(response.body.pageInfo).to.be.an('object');
+            expect(response.body.pageInfo.currentPage).to.eq('1');
+            expect(response.status).to.eq(200);
+            expect(response.body.customers.length).to.eq(10);
+            expect(response.body.pageInfo).to.be.an('object');
+            expect(response.body.pageInfo.currentPage).to.eq(response.body.pageInfo.currentPage);
```

Comment:
Essa _assertion_ vai sempre passar, pois você está comparando a mesma coisa.

Devia ser:

```suggestion
            expect(response.body.pageInfo.currentPage).to.eq(1);
```

Visto que `1` é passado como valor do parâmetro `page` durante a requisição.

> Isso vale para todos os testes de paginação.